storage/innobase/btr/btr0cur.cc

```diff
@@ -101,16 +101,16 @@ operations by purge as the previous, when it seems to be growing huge.
 throughput clearly from about 100000. */
 #define BTR_CUR_FINE_HISTORY_LENGTH	100000
 
+#ifdef BTR_CUR_HASH_ADAPT
 /** Number of searches down the B-tree in btr_cur_search_to_nth_level(). */
-Atomic_counter<ulint>	btr_cur_n_non_sea;
+ib_counter_t<ulint, ib_counter_element_t>	btr_cur_n_non_sea;
 /** Old value of btr_cur_n_non_sea.  Copied by
 srv_refresh_innodb_monitor_stats().  Referenced by
 srv_printf_innodb_monitor(). */
 ulint	btr_cur_n_non_sea_old;
-#ifdef BTR_CUR_HASH_ADAPT
 /** Number of successful adaptive hash index lookups in
 btr_cur_search_to_nth_level(). */
-ulint	btr_cur_n_sea;
+ib_counter_t<ulint, ib_counter_element_t>	btr_cur_n_sea;
 /** Old value of btr_cur_n_sea.  Copied by
 srv_refresh_innodb_monitor_stats().  Referenced by
 srv_printf_innodb_monitor(). */
```

Comment:
I think that changing the data type will break the code in `ha_innodb.cc`:
```c++
static SHOW_VAR innodb_status_variables[]= {
#ifdef BTR_CUR_HASH_ADAPT
  {"adaptive_hash_hash_searches", &btr_cur_n_sea, SHOW_SIZE_T},
  {"adaptive_hash_non_hash_searches", &btr_cur_n_non_sea, SHOW_SIZE_T},
#endif
```
You must extend `struct export_var_t` and `srv_export_innodb_status()` and make `innodb_status_variables[]` point to the added `export_vars` members that would cache a snapshot of the sharded counters.

We do not seem to have any regression test that would cover these counters. (No test failed because of this when I compiled with `cmake -DWITH_INNODB_AHI=OFF`.)